server/src/main/java/com/hedera/block/server/persistence/storage/write/BlockAsDirWriterBuilder.java

```diff
@@ -28,38 +29,37 @@
 import java.nio.file.attribute.FileAttribute;
 import java.nio.file.attribute.PosixFilePermission;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
 /**
  * Use builder methods to create a {@link BlockWriter} to write blocks to storage.
  *
  * <p>When a block writer is created, it will provide access to write blocks to storage.
  */
-public class BlockAsDirWriterBuilder {
-
+public final class BlockAsDirWriterBuilder {
     private final BlockNodeContext blockNodeContext;
-    private FileAttribute<Set<PosixFilePermission>> filePerms = FileUtils.defaultPerms;
+    private FileAttribute<Set<PosixFilePermission>> filePerms = DEFAULT_DIR_PERMISSIONS;
     private BlockRemover blockRemover;
 
     private BlockAsDirWriterBuilder(@NonNull final BlockNodeContext blockNodeContext) {
-        this.blockNodeContext = blockNodeContext;
-        PersistenceStorageConfig config =
+        this.blockNodeContext = Objects.requireNonNull(blockNodeContext);
+        final PersistenceStorageConfig config =
                 blockNodeContext.configuration().getConfigData(PersistenceStorageConfig.class);
-
         this.blockRemover =
-                new BlockAsDirRemover(Path.of(config.rootPath()), FileUtils.defaultPerms);
+                new BlockAsDirRemover(Path.of(config.rootPath()), DEFAULT_DIR_PERMISSIONS);
```

Comment:
Why are we setting permissions for removing folders?  That seems broken and incorrect.

Reply:
Good question, I am not sure as well. The file perms are used in a method inside the remover `remove(long)`. Based on the comments inside and the single usage it has currently, I believe the method is intended as a best-effort attempt to delete side effects caused by exceptions that could occur during the writing of a `BlockItem` to a file. There is a retry for the write which will first attempt to fix the file permissions and if it fails again then we go into the `remove(long)` method of the remover. Inside the remove method, the file perms that are passed in the constructor are used as `Files.setPosixFilePermissions(blockPath, filePerms.value());` before attempting to do the actual removal of the file. Maybe @mattp-swirldslabs could let us know and give us more context as to why this should be done like so?

Reply:
I would recommend just removing these permissions from the remover.
If we cannot delete a file, we also cannot change permissions (at least in any POSIX compliant system).

Reply:
✅ Done.

Reply:
I cannot say why the original set permissions for delete, but there is no possible scenario where it makes sense. We cannot set permissions in a way that changes whether we can remove the file.
There is no reason to be nervous at this stage, as we aren't deploying anything live and can always add things back if we later determine it is necessary.

I would strongly recommend removing that test (which was almost certainly added only to achieve a high coverage metric, rather than implementing a requirement or constraint) and removing the unnecessary value.

Side note, this is why sometimes less tests are better, we have code that someone is afraid to remove because it has a test, but both code and test are unequivocally unnecessary, as the value in question cannot be useful in any likely scenario.

This also highlights the value of a practice I've long advocated.  For every unit test scenario, document the test (or test case) with a clear description of the _requirement_ or _constraint_ being verified.  If a test does not verify a _requirement_ or _constraint_, then it should not exist.

Reply:
@jsync-swirlds thanks for the suggestion, I will remove it, to me also the functionality made no sense especially given the context that you provide in this thread. But I wanted to be absolutely sure nevertheless before doing it. :)

Reply:
✅ Done.